### PR TITLE
cmake: define KOKKOS_ARCH_A64FX

### PR DESCRIPTION
**Pull request by e10harvey**

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]
Somehow we add only added with the makefiles...

**Comment by e10harvey** [comment]
retest this please

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 1]
```diff
@@ -3063,7 +3063,9 @@ struct ViewValueFunctor<DeviceType, ValueType, true /* is_scalar */> {
                    std::is_trivially_copy_assignable<Dummy>::value>
   construct_shared_allocation() {
     // Shortcut for zero initialization
+#ifndef KOKKOS_ARCH_A64FX
     ValueType value{};
+#endif
 // On A64FX memset seems to do the wrong thing with regards to first touch
 // leading to the significant performance issues
 #ifndef KOKKOS_ARCH_A64FX
```
Comment (masterleinad) [inline core/src/impl/Kokkos_ViewMapping.hpp:3071]: Join the two `#ifdef`s.

**Review by masterleinad — CHANGES_REQUESTED** [review]

**Review by masterleinad — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]

**Comment by dalg24** [comment]
Ignoring CUDA failures due to issues with one of the machines

**Pull request merged**